tests/testthat/test-derive_param_tte.R

```diff
@@ -932,17 +932,203 @@ test_that("derive_param_tte Test 13: error if dataset_name not in source_datsets
   )
 })
 
+## Test 14: derive_param_tte detects duplicates when check_type = 'warning' ----
+test_that("derive_param_tte Test 14: detects duplicates in input datasets via pipeline functions", {
+  # Define ADSL dataset
+  adsl <- tibble::tribble(
+    ~USUBJID, ~TRTSDT,           ~TRTEDT,           ~EOSDT,
+    "01",     ymd("2020-12-06"), ymd("2021-03-02"), ymd("2021-03-06"),
+    "02",     ymd("2021-01-16"), ymd("2021-01-20"), ymd("2021-02-03")
+  ) %>% mutate(STUDYID = "AB42")
+
+  # Define AE dataset with duplicates
+  ae <- tibble::tribble(
+    ~USUBJID, ~AESTDTC,     ~AESEQ, ~AEDECOD,
+    "01",     "2021-01-03",      1, "Flu",
+    "01",     "2021-03-04",      2, "Cough",
+    "01",     "2021-01-03",      3, "Flu"
+  ) %>% mutate(
+    STUDYID = "AB42",
+    AESTDT = ymd(AESTDTC)
+  )
+
+  # Define event and censor sources
+  ttae <- event_source(
+    dataset_name = "ae",
+    date = AESTDT,
+    set_values_to = exprs(
+      EVENTDESC = "AE",
+      SRCDOM = "AE",
+      SRCVAR = "AESTDTC",
+      SRCSEQ = AESEQ
+    )
+  )
+
+  eot <- censor_source(
+    dataset_name = "adsl",
+    date = pmin(TRTEDT + days(10), EOSDT),
+    censor = 1,
+    set_values_to = exprs(
+      EVENTDESC = "END OF TRT",
+      SRCDOM = "ADSL",
+      SRCVAR = "TRTEDT"
+    )
+  )
+
+  # Run derive_param_tte and check for warning
+  expect_warning(
```

Comment:
`expect_snapshot()` should be used because the message is created by the function which is tested. This ensures that the complete message looks as intended.